### PR TITLE
steam-tui: init at 0.1.0

### PR DESCRIPTION
**Pull request by legendofmiracles**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 122699` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>steam-tui</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 122699` at 3a55b18e run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>steam-tui</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `makeWrapper`.
Near pkgs/games/steam-tui/default.nix:7:1:
```
  |
7 | makeWrapper,
  | ^
```
</li>
  </ul>
</details>

File: pkgs/games/steam-tui/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
+lib,
+rustPlatform,
+steamcmd,
+fetchFromGitHub,
+steam-run-native,
+runtimeShell,
+withWine ? false,
+wine
+}:
```
Comment (SuperSandro2000) [inline pkgs/games/steam-tui/default.nix:10]:
```suggestion
{ lib
, rustPlatform
, steamcmd
, fetchFromGitHub
, steam-run-native
, runtimeShell
, withWine ? false
, wine
}:
```

File: pkgs/games/steam-tui/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "steam-tui";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "dmadisetti";
+    repo = pname;
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/games/steam-tui/default.nix:19]:
```suggestion
    rev = version;
```

File: pkgs/games/steam-tui/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    sha256 = "sha256-UTXYlPecv0MVonr9zZwfwopfC/Fdch/ZSCxqgUsem40=";
+  };
+
+  cargoSha256 = "sha256-VYBzwDLSV4N4qt2dNgIS399T2HIbPTdQ2rDIeheLlfo=";
+
+  buildInputs = [ steamcmd steam-run-native ] ++ lib.lists.optional withWine wine;
```
Comment (SuperSandro2000) [inline pkgs/games/steam-tui/default.nix:25]:
```suggestion
  buildInputs = [ steamcmd steam-run-native ]
    ++ lib.optional withWine wine;
```

Comment (legendofmiracles) [reply pkgs/games/steam-tui/default.nix]: Merged

**Review by moni-dz — APPROVED** [review]

**Pull request merged**